[Bug] Find wrong Java Home for some distribution using sdkman

<img width="1342" alt="Screenshot 2023-10-07 at 11 01 19" src="https://github.com/Eskibear/node-jdk-utils/assets/6193897/928d73fc-d162-40fc-9c6e-f7508438979f">

If the Gradle Java Home is set to `/Users/sheche/.sdkman/candidates/java/8.0.372-zulu/zulu-8.jdk`, the project import will fail due to:

```
Caused by: org.gradle.internal.jvm.JavaHomeException: The supplied javaHome seems to be invalid. I cannot find the java executable. Tried location: /Users/sheche/.sdkman/candidates/java/8.0.372-zulu/zulu-8.jdk/bin/java
```

--- a/src/from/sdkman.ts
+++ b/src/from/sdkman.ts
@@ -2,7 +2,7 @@ import * as fs from "fs";
 import * as os from "os";
 import * as path from "path";
 import { log } from "../logger";
-import { isArm, isLinux, isMac } from "../utils";
+import { getRealHome, isArm, isLinux, isMac } from "../utils";
 
 const SDKMAN_DIR = process.env.SDKMAN_DIR ?? path.join(os.homedir(), ".sdkman");
 
@@ -26,7 +26,12 @@ export async function candidates(): Promise<string[]> {
         try {
             const files = await fs.promises.readdir(jdkBaseDir, { withFileTypes: true });
             const homedirs = files.filter(file => file.isDirectory()).map(file => path.join(jdkBaseDir, file.name));
-            ret.push(...homedirs);
+            // Resolve real Java Home by following bin/java. For some distributions (e.g. Zulu on macOS),
+            // bin/java is a symlink into a nested .../Contents/Home folder, and tools like the Gradle
+            // daemon reject the surface path as a mismatched JAVA_HOME. Falls back to the original path
+            // when bin/java cannot be resolved.
+            const resolved = await Promise.all(homedirs.map(async (h) => (await getRealHome(h)) ?? h));
+            ret.push(...resolved);
         } catch (error) {
             log(error);
         }

